engine/apps/integrations/views.py

```diff
@@ -115,27 +127,78 @@ def post(self, request):
 
                 create_alertmanager_alerts.apply_async((alert_receive_channel.pk, alert))
 
-        return Response("Ok.")
+    def process_v2(self, request, alert_receive_channel):
+        """
+        process_v2 creates one alert from one incoming AlertManager payload
+        """
+        alerts = request.data.get("alerts", [])
+
+        data = request.data
+        if "firingAlerts" not in request.data:
```

Comment:
consider adding a check for `max_alerts` (or limit alerts), to avoid processing alerts from misconfigured alertmanagers

Reply:
It won't help us. If we receive request which suits our size limits, iteration over array which already in memory is almost free